corehq/warehouse/models/facts.py

```diff
@@ -53,7 +53,7 @@ class FormFact(BaseFact, CustomSQLETLMixin):
     xmlns = models.CharField(max_length=255)
     user_id = models.CharField(max_length=255, null=True)
 
-    user_dim = models.ForeignKey(UserDim, on_delete=models.PROTECT)
+    user_dim = models.ForeignKey(UserDim, on_delete=models.PROTECT, null=True)
```

Comment:
how come this is nullable? weren't we using the not null constraint here to make the process fail hard if it didn't get a match in the join to user dim?

Reply:
There are forms with null users. Often submitted by the system. For the `FormFact` it seemed worth keeping them, in case they are needed in other reports later. They should not be used in the app status fact.